dallinger/utils.py

```diff
@@ -1,20 +1,69 @@
 from dallinger.config import get_config
+from functools import update_wrapper
 import os
 import random
 import string
 
 
 def get_base_url():
     config = get_config()
-    if 'OPENSHIFT_SECRET_TOKEN' in os.environ:
-        base_url = "http://{}/ad".format(config.get('host'))
+    host = os.getenv('HOST', config.get('host'))
+    if 'herokuapp.com' in host:
+        base_url = "https://{}".format(host)
     else:
-        base_url = "http://{}:{}/ad".format(
-            config.get('host'), config.get("port")
+        # debug mode
+        base_url = "http://{}:{}".format(
+            host, config.get("port")
         )
     return base_url
 
 
 def generate_random_id(size=6, chars=string.ascii_uppercase + string.digits):
     ''' Generate random id numbers '''
     return ''.join(random.choice(chars) for x in range(size))
+
+
+class reify(object):
```

Comment:
This is a tricky one. David pointed me to `reify` after reviewing my awkward in-place implementation and recognizing that it was what I was looking for. This kind of memoizing decorator is handy to have, so I doubt this would be the only place it gets used, but maybe that's a premature assumption.

It seems like the borrowing happening here is more just Python expertise than Pyramid, and maybe it's fine do just grab it without attribution. I included the attribution more because of the docstring than the code. Hate to bother you @davisagli, but I'd love your opinion here.